rpcs3/Emu/Cell/Modules/sceNp.cpp

```diff
@@ -2362,24 +2512,28 @@ error_code sceNpManagerGetNpId(ppu_thread& ppu, vm::ptr<SceNpId> npId)
 		return SCE_NP_ERROR_INVALID_ARGUMENT;
 	}
 
-	if (g_psn_connection_status == SCE_NP_MANAGER_STATUS_OFFLINE)
+	if (nph->get_psn_status() == SCE_NP_MANAGER_STATUS_OFFLINE)
 	{
 		return not_an_error(SCE_NP_ERROR_OFFLINE);
 	}
 
-	if (g_psn_connection_status != SCE_NP_MANAGER_STATUS_LOGGING_IN && g_psn_connection_status != SCE_NP_MANAGER_STATUS_ONLINE)
+	if (nph->get_psn_status() != SCE_NP_MANAGER_STATUS_LOGGING_IN && nph->get_psn_status() != SCE_NP_MANAGER_STATUS_ONLINE)
 	{
 		return SCE_NP_ERROR_INVALID_STATE;
 	}
 
+	memcpy(npId.get_ptr(), &nph->get_npid(), sizeof(SceNpId));
```

Comment:
Nitpick, but `sizeof(&nph->get_npid())` may work. Or something similar, to tie the size to this function's return type.

Reply:
Same^